### DISCUSSION TITLE
blocked by CORS policy

### DISCUSSION THREAD

**Discussion by MSYXing** [reworkd/AgentGPT#638] [Q&A] [answered]
On version 0.5, I tried running this command (docker-compose -f deploy/docker-compose.yml --project-directory . up --build) and it showed that it was started, but the request (http://localhost:3000/api/agent/start) showed cross-domain problems, and then I added this link(https://cors-anywhere.herokuapp.com/http://localhost:3000/api/agent/start) to fix it, but now 403

**Comment by Borozenets** [comment]
Hi @MSYXing 
I had the same problem.
I was helped by experimenting with the "frontend_url:" variable, which is located in the path 
AgentGPT/platform/reworkd_platform/settings.py
It probably helps you.
Block:
    # Frontend URL for CORS
    frontend_url: str = "http://localhost:3000"

**Reply by MSYXing** [reply]
Can you be more specific?

**Reply by Borozenets** [reply, marked as answer]
I tested on  my Raspberry Pi, and in my case I don't need security as a test.  My domain for the PI test, as an example http://pi:3000
An example of my configs:

.env 
---
root@pi:~/AgentGPT# cat next/.env | grep http://pi
NEXT_PUBLIC_BACKEND_URL="http://pi:8000"
NEXTAUTH_URL="http://pi:3000"
REWORKD_PLATFORM_FRONTEND_URL="http://pi:3000"
---

settings.py
---
root@pi:~/AgentGPT# cat ../AgentGPT/platform/reworkd_platform/settings.py | grep http://pi
    frontend_url: str = "http://192.168.1.35:3000,http://localhost,http://localhost:3000,http://pi:3000,http://pi"
---

**Reply by MSYXing** [reply]
Thank you for your reply. I have modified the configuration according to what you said. The problems of CORS and 403 have been solved, but now the request is returned to 409 Conflict

**Reply by Borozenets** [reply]
Oh, I'll think about it, I didn't have such an error.
But to be honest, I don't remember 409 at all in my practice
